### PR TITLE
multiple trivial fixlets, across the tree

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/fsck/fsck.c [hunk 1]
```diff
@@ -414,7 +414,7 @@ static int run(int argc, char *argv[]) {
         exit_status = wait_for_terminate_and_check("fsck", pid, WAIT_LOG_ABNORMAL);
         if (exit_status < 0)
                 return exit_status;
-        if (exit_status & ~1) {
+        if ((exit_status & ~FSCK_ERROR_CORRECTED) != 0) {
```
Comment (keszybz) [inline src/fsck/fsck.c]: Do we really need the `!= 0` part? I think it just distracts.

File: src/shared/json.c [hunk 2]
```diff
@@ -3140,10 +3140,7 @@ int json_log_internal(
         va_list ap;
         int r;
 
-        if (error < 0)
-                error = -error;
-
-        errno = error;
+        errno = ERRNO_VALUE(error);
```
Comment (keszybz) [inline src/shared/json.c:3145]: Oops.

File: src/fsck/fsck.c (hunk 1 above)
Comment (poettering) [reply src/fsck/fsck.c]: i was tempted to, but our coding style we should drop it only for boolean-like stuff, but this is kinda not so much anymore, after all we mask out the "corrected" flag and leave all the other bits on, which are not booleans as a whole...

but actually, looking at this, I think this should be rewritten as:

```c
if ((exit_status & ~FSCK_ERROR_CORRECTED) == FSCK_SUCCESS)) …
```

**Comment by poettering** [comment]
I force pushed a new version now. two changes: i made the FSCK_SUCCESS change i mentioned above. I also added one more commit  on top (last one in the series) that changes all remaining bitfield enum definitions to use the "1 << x" syntax. tried to be careful with the latter so that nothing actually effectively changed, but maybe somebody else should have another look on that, hence dropping the green label for now. ptal!

**Pull request merged**